data/src/main/java/com/my/kizzy/data/utils/Ext.kt

```diff
@@ -36,7 +38,29 @@ import io.ktor.http.HttpStatusCode
 import java.io.File
 import java.io.FileOutputStream
 
-suspend fun HttpResponse.toImageAsset(): String? {
```

Comment:
Can you try to make it a generic function which handles different body types. I think something like 

```kt
suspend inline fun <reified T : Any> HttpResponse.toImageAsset(): String? {
    return try {
        if (this.status == HttpStatusCode.OK) {
            when (T::class) {
                ImgurResponse::class -> (this.body<T>() as ImgurResponse).data.link
                Array<ExternalAsset>::class -> (this.body<T>() as Array<ExternalAsset>)
                    .firstOrNull()?.let { "mp:" + it.externalAssetPath }
                ApiResponse::class -> (this.body<T>() as ApiResponse).id
                else -> throw UnsupportedOperationException("Unsupported type: ${T::class}")
            }
        } else {
            null
        }
    } catch (e: Exception) {
        null
    }
}
```

If it becomes too complex/messy we can have 3 different methods I don't have a problem with that

Reply:
call it as 
```kt
val imageUrl: String? = response.toImageAsset<ImgurResponse>()
val externalAssetUrl: String? = response.toImageAsset<Array<ExternalAsset>>()
val attachmentAssetUrl: String? = response.toImageAsset<ApiResponse>()
```

Reply:
> If it becomes too complex/messy we can have 3 different methods I don't have a problem with that

This already looks really messy for a simple method tbh.
Imgur URL and Asset conversion doesn't do the same thing either so semantically it doesn't make sense to call the same generic function.

Reply:
cool